src/main/java/com/intuit/graphql/orchestrator/federation/EntityDataFetcher.java

```diff
@@ -12,19 +12,26 @@
 import graphql.schema.DataFetcher;
 import graphql.schema.DataFetchingEnvironment;
 import java.util.ArrayList;
-import java.util.Collections;
+import java.util.Collection;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.CompletableFuture;
-import java.util.stream.Collectors;
 import lombok.RequiredArgsConstructor;
 import org.apache.commons.collections4.CollectionUtils;
 
+/**
+ * This class is used for resolving fields added to an Entity by making
+ * an entity fetch request.  To build the entity fetch query, it uses
+ * {@link EntityQuery}.
+ */
 @RequiredArgsConstructor
 public class EntityDataFetcher implements DataFetcher<CompletableFuture<Object>> {
 
+  private static final Field __typenameField =
```

Comment:
move to introspection util. I think we are using it somewhere else also, for interface/unions

Reply:
good catch